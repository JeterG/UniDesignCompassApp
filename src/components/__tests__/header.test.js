import React from "react";
import renderer from "react-test-renderer";
import {render, fireEvent, wait} from 'react-testing-library';
import Header from "../header";
import 'react-testing-library/cleanup-after-each';

describe("Header", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<Header siteTitle="Universal Design Compass" />)
      .toJSON()
    expect(tree).toMatchSnapshot();
  });
  it("select about", () => {
    // const { getByText, getByTestId, container } = render(<Header siteTitle="Universal Design Compass" />);
    // fireEvent.click(getByText("About Us"));
    // expect(container.firstchild).toMatchSnapshot();
    const { getByText } = render(<Header siteTitle="Universal Design Compass" />);
    const aboutButton = getByText("About Us");
    const textAbout = aboutButton.textContent;
    expect(textAbout).toBe("About Us");
    aboutButton.click();
  });
});
