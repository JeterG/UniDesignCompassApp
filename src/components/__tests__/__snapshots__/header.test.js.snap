// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<nav
  className="navbar navbar-expand-md navbar-dark bg-dark sticky-top"
>
  <span
    className="navbar-brand"
    variant="light"
  >
     
    <a
      href="/"
    >
       
      Universal Design Compass
    </a>
     
  </span>
  <button
    aria-controls="basic-navbar-nav"
    aria-label="Toggle navigation"
    className="navbar-toggler"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    aria-expanded={null}
    className="navbar-collapse collapse"
    id="basic-navbar-nav"
  >
    <div
      className="navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        className="nav-link"
        data-rb-event-key={null}
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
      >
        <a
          href="/"
        >
          About Us
        </a>
      </a>
      <a
        className="nav-link"
        data-rb-event-key={null}
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
      >
        <a
          href="/"
        >
          Contact
        </a>
      </a>
      <a
        className="nav-link"
        data-rb-event-key={null}
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
      >
        <a
          href="/"
        >
          Create
        </a>
      </a>
      <a
        className="nav-link"
        data-rb-event-key={null}
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
      >
        <a
          href="/Compass"
        >
          Compass
        </a>
      </a>
      <a
        className="nav-link"
        data-rb-event-key={null}
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
      >
        <a
          href="/Login"
        >
          Login
        </a>
      </a>
    </div>
  </div>
</nav>
`;
